### ISSUE TITLE
Question about the loss computation

### ISSUE THREAD

**Issue by da0c** [Jongchan/tensorflow-vdsr#21]
Hello, I just checking training process in my case. 
So, I look in the loss computation for weight decay term:
`
for w in weights:
     loss += tf.nn.l2_loss(w)*1e-4
`
this is always about 0.002 in my case.
Are you shure that this computation is correct?

**Comment by da0c** [comment]
It is all right with weight decay computation here, the mistake was on my side.
In my case increasing weight decay multiplayer to 0.01 allows to improve network generalization.
Thank you for this nice VDSR implementation!

**Comment by wwlCape** [comment]
hi,
I have some confuse about the work of   "for w in weights: loss += tf.nn.l2_loss(w)*1e-4".Could you tell me the reason to add this code?   Looking forward to hear your reply! @da0c

**Comment by da0c** [comment]
Hi, it implements the weight decay regularization. It is necessary for the first iterations of training process.
UPDATE - Weight decay applies limitation to the CNN weights, it works similar to Tikhonov regularization - https://en.wikipedia.org/wiki/Tikhonov_regularization
